Bluepill-cli/Bluepill-cli/Simulator/SimulatorMonitor.m

```diff
@@ -135,6 +137,10 @@ - (void)onTestCaseFailedWithName:(NSString *)testName inClass:(NSString *)testCl
      [currentTime timeIntervalSinceDate:self.lastTestCaseStartDate],
      fullTestName];
 
+    // Save screenshot for failed test
+    NSString *screenshotName = [fullTestName stringByReplacingOccurrencesOfString:@"/" withString:@"_"];
+    [self.screenshotService saveScreenshotForFailedTestWithName:screenshotName];
```

Comment:
it is too late to take screenshot now.. Don't you want to take a screenshot right after XCTFail() or Assertions instead of at end of the test?